lib/ocpp/v16/smart_charging.cpp

```diff
@@ -375,16 +375,19 @@ void SmartChargingHandler::clear_all_profiles() {
     }
 }
 
-std::vector<ChargingProfile> SmartChargingHandler::get_valid_profiles(const ocpp::DateTime& start_time,
-                                                                      const ocpp::DateTime& end_time,
-                                                                      const int connector_id) {
+std::vector<ChargingProfile>
+SmartChargingHandler::get_valid_profiles(const ocpp::DateTime& start_time, const ocpp::DateTime& end_time,
+                                         const int connector_id,
+                                         const std::set<ChargingProfilePurposeType>& purposes_to_ignore) {
     std::vector<ChargingProfile> valid_profiles;
 
     {
         std::lock_guard<std::mutex> lk(charge_point_max_profiles_map_mutex);
 
-        for (const auto& [stack_level, profile] : stack_level_charge_point_max_profiles_map) {
-            valid_profiles.push_back(profile);
+        if (not purposes_to_ignore.count(ChargingProfilePurposeType::ChargePointMaxProfile)) {
```

Comment:
observation
I would consider using EnumFlags here rather than a set.
For a set (or other container based solution) count() has the disadvantage that it doesn't stop on finding the first matching entry. using find() would be better.
Also count() returns a size type rather than a boolean; checking `== 0` or `!= 0` is more appropriate than using `not`.
e.g.
```
if (std::find(
    std::begin(purposes_to_ignore), 
    std::end(purposes_to_ignore), 
    ChargingProfilePurposeType::ChargePointMaxProfile) != std::end(purposes_to_ignore)) {...}
```
or
```
if (auto search = purposes_to_ignore.find(ChargingProfilePurposeType::ChargePointMaxProfile); search != purposes_to_ignore.end()) {...}
```